tensorflow/python/training/adadelta_test.py

```diff
@@ -20,104 +20,90 @@
-            var0.eval(),
-            rtol=1e-3)
-
-        self.assertAllCloseAccordingToType(
-            np.array([3.0 - update1 - update2, 4.0 - update1 - update2]),
-            var1.eval(),
-            rtol=1e-3)
-
+        for grad in [0.2, 0.1, 0.01]:
+            for lr in [1.0, 0.5, 0.1]:
+              with self.test_session():
+                var0_init = [1.0, 2.0]
+                var1_init = [3.0, 4.0]
+                var0 = tf.Variable(var0_init, dtype=dtype)
+                var1 = tf.Variable(var1_init, dtype=dtype)
+
+                grads = tf.constant([grad, grad], dtype=dtype)
+
+                accum = 0.0
+                accum_update = 0.0
+
+                # ADADELTA gradient optimizer
+                rho = 0.95
+                epsilon = 1e-8
+                adadelta_opt = tf.train.AdadeltaOptimizer(lr, rho, epsilon)
+                adadelta_update = adadelta_opt.apply_gradients(zip(
+                    [grads, grads], [var0, var1]))
+
+                tf.initialize_all_variables().run()
+
+                # Assign slots
+                slot = [None]*2
+                slot_update = [None]*2
+                self.assertEqual(["accum", "accum_update"],
+                                 adadelta_opt.get_slot_names())
+                slot[0] = adadelta_opt.get_slot(var0, "accum")
+                self.assertEquals(slot[0].get_shape(), var0.get_shape())
+                self.assertFalse(slot[0] in tf.trainable_variables())
+
+                slot_update[0] = adadelta_opt.get_slot(var0, "accum_update")
+                self.assertEquals(slot_update[0].get_shape(), var0.get_shape())
+                self.assertFalse(slot_update[0] in tf.trainable_variables())
+
+                slot[1] = adadelta_opt.get_slot(var1, "accum")
+                self.assertEquals(slot[1].get_shape(), var1.get_shape())
+                self.assertFalse(slot[1] in tf.trainable_variables())
+
+                slot_update[1] = adadelta_opt.get_slot(var1, "accum_update")
+                self.assertEquals(slot_update[1].get_shape(), var1.get_shape())
+                self.assertFalse(slot_update[1] in tf.trainable_variables())
+
+                # Fetch params to validate initial values
+                self.assertAllClose(var0_init, var0.eval())
+                self.assertAllClose(var1_init, var1.eval())
+
+                update = [None]*num_updates
+                tot_update = 0
+                for run_no in range(num_updates):
+                    # Run adadelta update for comparison
+                    adadelta_update.run()
+
+                    # Perform initial update without previous accum values
+                    accum = accum * rho + (grad**2) * (1 - rho)
+                    update[run_no] = (np.sqrt(accum_update + epsilon) *
+                        (1. / np.sqrt(accum + epsilon)) * grad)
+                    accum_update = accum_update * rho + (update[run_no]**2) * (1.0 - rho)
```

Comment:
our linters go nuts when lines go over 80 characters, can you try to wrap these to 80 cols as necessary?)